src/main/java/frc/robot/subsystems/Indexer.java

```diff
@@ -115,17 +116,18 @@ public CommandBase reverseOutput() {
         return indexMotor(-PIERRE_INDEX_SPEED);
     }
 
-    // This command will make sure that the singulator has possesion of the ball
-
+    // run singulator when front intake is triggered, stop singulator when back
+    // intake IR is clear and until bottom pierre is covered
+    // run pierre when bottom pierre is covered
     public CommandBase pierrePossesion() {
         CommandBase cmd = new FunctionalCommand(() -> {
         }, () -> {
-            if (frontIntakeIR.getAsBoolean()) {
+            if (frontIntakeIR.getAsBoolean() || backIntakeIR.getAsBoolean() && !topPierreIR.getAsBoolean()) {
```

Comment:
is this `(a||b) && c`, or `a || (b && c)`? Use parenthesis.

Reply:
resolved